src/h5web/providers/models.ts

```diff
@@ -48,6 +48,47 @@ export interface HDF5Datatype<T = HDF5Type> {
   type: T;
 }
 
+export enum MyHDF5EntityKind {
+  Group = 'group',
+  Dataset = 'dataset',
+  Datatype = 'datatype',
+  Link = 'link',
+}
+
+export interface MyHDF5Entity {
+  uid: string;
+  id?: HDF5Id;
+  name: string;
+  kind: MyHDF5EntityKind;
+  parents: MyHDF5Group[];
+  attributes: HDF5Attribute[];
+  rawLink: HDF5Link;
+  rawEntity?: HDF5Entity;
+}
```

Comment:
The new models (with temporary names). The old models may eventually be replaced or moved to `SilxProvider`.

Note that apart from being nested, a big change is that `MyHDF5Entity` is no longer a union but a base class that other interfaces extend. I read this tip somewhere recently on an article about optimising the performance of TypeScript: with this object-oriented pattern, when we use the base type `MyHDF5Entity`, the engine knows right away which keys are allowed; it doesn't have to look for common keys across all the members of the union. I'll share the article on #webdev. I think this pattern is also more common in the wild, and it feels more descriptive than a simple union (the common keys are clearly identified).